### PR TITLE
Add --only-matching option

### PR DESCRIPTION
**Pull request by hi-artem**
Add only-matching option to only output matching parts of the string (similar to how rg goes it).

I had to replace `printer::Standard` with `printer::StandardBuilder` in order to achieve that - hope it's not an issue.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -56,6 +56,7 @@ secrets into version control.
 USAGE:
     ripsecrets [--strict-ignore] [PATH ...]
     ripsecrets --install-pre-commit
+    ripsecrets --only-matching
```
Comment (sirwart) [inline src/main.rs:59]: This should be an option on line 57 instead of a new line

**Review by sirwart — COMMENTED** [review]
I like this feature but it doesn't quite work for random strings unfortunately. The reason why is the random string regex in the form of `key = "<secret>"`, so when --only-matching is invoked it will output the entire `key = "<secret>"` instead of just `<secret>`. There's special code to handle the difference in the regexes for secrets that are whitelisted in the `.secretsignore` file. I think the start position of the match needs to be updated when the random string regex is hit, which is tested on this line https://github.com/sirwart/secrets/blob/main/src/matcher.rs#L92

**Comment by hi-artem** [comment]
@sirwart Ah yea i see that now. I couldn't wrap my head around the logic in `find_at`. Will spend more time on it later.

**Comment by sts10** [comment]
Hope you won't have much trouble adapting this change to the new argument parsing structure, which uses [clap](https://docs.rs/clap/latest/clap/index.html). 

You'll likely want to do something like this in `src/main.rs`:

```rust
struct Args {
    /// Install `ripsecrets` as a pre-commit hook automatically in git directory provided. Defaults to
    /// '.'
    #[clap(long = "install-pre-commit")]
    install_pre_commit: bool,

    /// If you pass a path as an argument that's ignored by .secretsignore it
    /// will be scanned by default. --strict-ignore will override this
    /// behavior and not search the paths passed as arguments that are excluded
    /// by the .secretsignore file. This is useful when invoking secrets as a
    /// pre-commit.
    #[clap(long = "strict-ignore")]
    strict_ignore: bool,

    /// Print only the matched (non-empty) parts of a matching line, with each such
    /// part on a separate output line
    #[clap(long = "only-matching")]
    only_matching: bool,

    /// Source files. Can be files or directories. Defaults to '.'
    #[clap(name = "Source files", parse(from_os_str))]
    paths: Vec<PathBuf>,
}
```

And then, further down in `main.rs`, I assume you'll want to pass the bool to `find_secrets`. That might look something like:

```rust
match find_secrets(&paths, args.strict_ignore, args.only_matching) {
    // ...
}
```

Feel free to tag me if you have questions!

**Comment by sirwart** [comment]
@hi-artem I added this option in a0eedb9ea855b2a4445231b68d63fe7240afaf89 since it needed to be adapted to the new argument parsing library and I fixed the issue with the output for random string matches. Thanks again for suggesting this and opening this PR!

**Comment by hi-artem** [comment]
@sirwart that's awesome! i have never got time to look the pr again, and glad to her the option was added!

**Pull request closed without merging**